packages/relay/index.js

```diff
@@ -57,7 +57,9 @@ type CommitMutationConfig = {|
 export type QueryRendererProps = {|
   +query: string,
   +render: (props: Object) => React.Node,
+  +renderOfflineScreen?: ?(retry: () => void) => React.Node,
```

Comment:
Why are these optional props also nullable/undefined?

Reply:
This is the issue with exporting/sharing prop types.

Since they QueryRendererProps is used by PrivateApiRenderer, PublicApiRenderer and QueryRenderer. The prop is optional when passing it to PrivateApiRenderer and PublicApiRenderer, but they both pass it along to QueryRenderer, so it will always be passed to QueryRenderer, but it could be undefined.